### PR TITLE
virtcontainers/pkg/compatoci/: Improved coverage for  for Kata 2.0

### PR DESCRIPTION
**Pull request by Liu-Weijie**
In this PR, I add some test to expand test coverage. Now the coverage of compatoci is 91.5%

Credit PR to Hackathon Team3

Fixes: #258

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @Liu-Weijie.

lgtm

File: src/runtime/virtcontainers/pkg/compatoci/utils_test.go [hunk 1]
```diff
@@ -145,3 +179,55 @@ func TestGetConfigPath(t *testing.T) {
 	configPath := getConfigPath(tempBundlePath)
 	assert.Equal(t, configPath, expected)
 }
+
+func TestParseConfigHSON(t *testing.T) {
```
Comment (amshinde) [inline src/runtime/virtcontainers/pkg/compatoci/utils_test.go:183]: nit: Seems like a typo here. Should be `TestParseConfigJSON` here.

File: src/runtime/virtcontainers/pkg/compatoci/utils_test.go [hunk 2]
```diff
@@ -97,6 +111,26 @@ func TestContainerCapabilities(t *testing.T) {
 	assert.Equal(t, c.Permitted, []string(nil))
 	assert.Equal(t, c.Inheritable, []string(nil))
 	assert.Equal(t, c.Ambient, []string(nil))
+
+	ociSpec.Process.Capabilities = interface{}("")
+
+	c, err = ContainerCapabilities(ociSpec)
+	assert.NotNil(t, err)
+	assert.Equal(t, c.Bounding, []string(nil))
```
Comment (amshinde) [inline src/runtime/virtcontainers/pkg/compatoci/utils_test.go:119]: This check is fine, but a better way would be to check if len(c.Bounding) is 0. Similarly for other checks as well.

File: src/runtime/virtcontainers/pkg/compatoci/utils_test.go (hunk 1 above)
Comment (Liu-Weijie) [reply src/runtime/virtcontainers/pkg/compatoci/utils_test.go:183]: Thanks, I will fix it

File: src/runtime/virtcontainers/pkg/compatoci/utils_test.go (hunk 2 above)
Comment (Liu-Weijie) [reply src/runtime/virtcontainers/pkg/compatoci/utils_test.go:119]: Thanks, I will fix it

**Comment by amshinde** [comment]
@Liu-Weijie Thanks for the changes. lgtm. Could you squash the 2 commits into a single commit? This can be merged then.

**Comment by Liu-Weijie** [comment]
> @Liu-Weijie Thanks for the changes. lgtm. Could you squash the 2 commits into a single commit? This can be merged then.

Sure

**Review by amshinde — APPROVED** [review]

**Comment by chavafg** [comment]
@Liu-Weijie could you please address the static checks failure?
```
utils_test.go:185: File is not `gofmt`-ed with `-s` (gofmt)
	
make: *** [Makefile:41: static-checks] Error 1
```

**Comment by Liu-Weijie** [comment]
> @Liu-Weijie could you please address the static checks failure?
> 
> ```
> utils_test.go:185: File is not `gofmt`-ed with `-s` (gofmt)
> 	
> make: *** [Makefile:41: static-checks] Error 1
> ```

Fixed

**Pull request merged**